acceptance: better debugging for TestDecommission

tbg opened the pull request:
A test hang was observed in CI where the `ExecCLI` never returned until the test was killed.
Make sure that if that were to repeat, more information would become available.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/19604)
<!-- Reviewable:end -->

tbg committed:
acceptance: better debugging for TestDecommission

A test hang was observed in CI where the `ExecCLI` never returned until the test was killed.
Make sure that if that were to repeat, more information would become available.
